Fix vertical scroll/zoom for precision touchpad on windows

georgwacker opened the pull request:
This fix adds the factor data for the vertical mousewheel event, much like the horizontal mousewheel event, where this is already present.

With this fix, vertical scroll/zoom with a precision touchpad on windows is smooth and allows for fine grain control.
The code to support precision touchpad input is already present, but not used correctly.

Mousewheel events for other devices are not affected and are functioning as expected (tested with regular mouse).

It would be great if this could be cherrypicked for stable.

Fixes #51313

akien-mga commented on platform/windows/display_server_windows.cpp:
```diff
@@ -2442,8 +2442,10 @@ LRESULT DisplayServerWindows::WndProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARA
+						mb->set_factor(fabs((double)motion / (double)WHEEL_DELTA));
 					} else {
 						mb->set_button_index(MOUSE_BUTTON_WHEEL_DOWN);
+						mb->set_factor(fabs((double)motion / (double)WHEEL_DELTA));
```
If it's the same call for both branches, it can go outside the `if`.

georgwacker replied:
Right, I cleaned up the code for the horizontal mousewheel event as well now. Do I need to squash the commits?

akien-mga replied:
Yes, please do.

georgwacker committed:
Fix vertical scroll/zoom for precision touchpad

Adds factor data for vertical mousewheel event, in line with horizontal mousewheel event

akien-mga commented:
Thanks!

akien-mga commented:
Cherry-picked for 3.4.

akien-mga commented:
Cherry-picked for 3.3.3.
